### PR TITLE
[Enhancement] support lake_clear_corrupted_cache option for BE

### PR DESCRIPTION
**Pull request by starrocks-xupeng**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.4
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0

### CODE REVIEW

File: be/src/storage/lake/tablet_manager.cpp [hunk 1]
```diff
@@ -243,7 +244,30 @@ StatusOr<TabletMetadataPtr> TabletManager::load_tablet_metadata(std::shared_ptr<
     auto t0 = butil::gettimeofday_us();
     auto metadata = std::make_shared<TabletMetadataPB>();
     ProtobufFile file(metadata_location, std::move(fs));
-    RETURN_IF_ERROR(file.load(metadata.get(), fill_cache));
+    auto s = file.load(metadata.get(), fill_cache);
```
Comment (wyb) [inline be/src/storage/lake/tablet_manager.cpp:247]: `segment` `delvec` files also have the same issue？

File: be/src/storage/lake/tablet_manager.cpp [hunk 2]
```diff
@@ -243,7 +244,30 @@ StatusOr<TabletMetadataPtr> TabletManager::load_tablet_metadata(std::shared_ptr<
...
+    if (!s.ok()) {
+        if (s.is_corruption() && config::lake_clear_corrupted_cache) {
```
Comment (wyb) [inline be/src/storage/lake/tablet_manager.cpp:249]: better to add info log

**Review by wyb — DISMISSED** [review]

File: be/src/storage/lake/tablet_manager.cpp (hunk 1 above)
Comment (starrocks-xupeng) [reply be/src/storage/lake/tablet_manager.cpp:247]: currently only found the problem in meta, so deal with meta only now

File: be/src/storage/lake/tablet_manager.cpp (hunk 2 above)
Comment (starrocks-xupeng) [reply be/src/storage/lake/tablet_manager.cpp:249]: done

**Review by wyb — APPROVED** [review]

**Review by kevincai — COMMENTED** [review]
add UT to cover the auto clean case.

**Review by kevincai — APPROVED** [review]

**Comment by starrocks-xupeng** [comment]
https://github.com/StarRocks/starrocks/pull/54250，branch-3.3

**Pull request merged**